app/components/post_body/post_body.js

```diff
@@ -390,14 +390,26 @@ export default class PostBody extends PureComponent {
         } else if (postType === Posts.POST_TYPES.COMBINED_USER_ACTIVITY) {
             const {allUserIds, allUsernames, messageData} = postProps.user_activity;
             messageComponent = (
-                <CombinedSystemMessage
+                <SystemMessage
                     allUserIds={allUserIds}
                     allUsernames={allUsernames}
                     linkStyle={textStyles.link}
                     messageData={messageData}
                     navigator={navigator}
                     textStyles={textStyles}
                     theme={theme}
+                    postType={postType}
+                />
+            );
+        } else if (postType === Posts.POST_TYPES.PURPOSE_CHANGE || postType === Posts.POST_TYPES.HEADER_CHANGE ||
```

Comment:
Instead of listing the types supported by `SystemMessage` here, we could make a static method on like `SystemMessage.supportsPostType` to avoid having to repeat the types in multiple places

Reply:
Thoughts on the above comment?

Reply:
I don't think that it would be a good idea. We would still have to pass post type to compare against since different post types are handled differently.

Reply:
You'll still have to pass in the post type, but it saves you from having to list the types repeatedly if you do something like
```javascript
const systemMessageRenderes = {
    [Posts.POST_TYPES.PURPOSE_CHANGE]: renderPurposeChange,
};

export default class SystemMessage extends React.PureComponent {
    render() {
         return systemMessageRenderers[this.props.post.type](this.props.post);
    }

    static supportsPostType(postType) {
        return Boolean(systemMessageRenderers[postType]);
    }
}
```
since then you could just have the `PostBody` render like this
```javascript
if (SystemMessage.supportsPostType(post.type)) {
    return <SystemMessage post={post}/>
}
```